### PR TITLE
More cleanup & modernization, notably ARC & nullability

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: Sources/Main/FRFeedbackReporter.m [hunk 1]
```diff
@@ -188,11 +178,10 @@ - (BOOL) reportException:(NSException *)exception
         
         [controller setSubheading:FRLocalizedString(@"Send crash report", nil)];
 
-        NSString* callStack = [exception my_callStack];
-        [controller setException:[NSString stringWithFormat: @"%@\n\n%@\n\n%@",
+        [controller setException:[NSString stringWithFormat: @"%@\n\n%@\n\n%@\n",
                                     [exception name],
                                     [exception reason],
-                                    callStack ? callStack : @""]];
+                                    [exception callStackSymbols]]];
```
Comment (tcurdt) [inline Sources/Main/FRFeedbackReporter.m]: Wow - since when does that exist?

File: Sources/Main/FRCrashLogFinder.m [hunk 2]
```diff
@@ -44,7 +46,7 @@ + (BOOL)file:(NSString*)path isNewerThan:(NSDate*)date
     return YES;
 }
 
-+ (NSArray*) findCrashLogsSince:(NSDate*)date
++ (NSArray*) findCrashLogsSince:(nullable NSDate*)date
```
Comment (tcurdt) [inline Sources/Main/FRCrashLogFinder.m]: a nullable input parameter feels wrong here

File: Sources/Main/FRCrashLogFinder.m [hunk 3]
```diff
@@ -19,8 +19,10 @@
 
 @implementation FRCrashLogFinder
 
-+ (BOOL)file:(NSString*)path isNewerThan:(NSDate*)date
++ (BOOL)file:(NSString*)path isNewerThan:(nullable NSDate*)date
```
Comment (tcurdt) [inline Sources/Main/FRCrashLogFinder.m]: a nullable input parameter feels wrong here

File: Sources/Main/FRCrashLogFinder.h [hunk 4]
```diff
@@ -16,9 +16,12 @@
 
 #import <Cocoa/Cocoa.h>
 
+NS_ASSUME_NONNULL_BEGIN
 
 @interface FRCrashLogFinder : NSObject
 
-+ (NSArray*) findCrashLogsSince:(NSDate*)date;
++ (NSArray*) findCrashLogsSince:(nullable NSDate*)date;
```
Comment (tcurdt) [inline Sources/Main/FRCrashLogFinder.h]: a nullable input parameter feels wrong here

File: Sources/Main/FRFeedbackReporter.m (hunk 1 above)
Comment (tcurdt) [inline Sources/Main/FRFeedbackReporter.m]: I see - since 10.6 :)

**Comment by tcurdt** [comment]
Looking good. Just the three nullable input parameters that seem a bit odd. That said: maybe they are required like that? I didn't check that yet.

File: Sources/Main/FRCrashLogFinder.m (hunk 3 above)
Comment (seanm) [inline Sources/Main/FRCrashLogFinder.m]: just a few lines down, it checks if the parameter is nil, so presumably it is expected/allowed.

I've got a big refactoring of FRCrashLogFinder coming though (using newer API).

File: Sources/Main/FRCrashLogFinder.h (hunk 4 above)
Comment (seanm) [inline Sources/Main/FRCrashLogFinder.h]: this is the only public API of this class, and the two callers pass a date from defaults, and there's always a chance that defaults doesn't have the sought key.

**Pull request merged**